src/diffusers/models/model_loading_utils.py

```diff
@@ -61,6 +69,21 @@ def _determine_device_map(model: torch.nn.Module, device_map, max_memory, torch_
     return device_map
 
 
+def _fetch_remapped_cls_from_config(config, old_class):
+    previous_class_name = old_class.__name__
+    remapped_class_name = _CLASS_REMAPPING_DICT.get(previous_class_name).get(config["norm_type"])
+
+    # load diffusers library to import compatible and original scheduler
+    diffusers_library = importlib.import_module(__name__.split(".")[0])
+    remapped_class = getattr(diffusers_library, remapped_class_name)
+    logger.info(
+        f"Changing class object to be of `{remapped_class_name}` type from `{previous_class_name}` type."
```

Comment:
Just a nit. Maybe add a sentence about why the remapping is being done. Something like `Transformer2DModel is scheduled to be deprecated in a future version"?